semanticcpg/src/main/scala/io/shiftleft/semanticcpg/sarif/SarifSchema.scala

```diff
@@ -0,0 +1,315 @@
+      */
+    def codeFlows: List[CodeFlow]
+  }
+
+  /** Describes a single run of an analysis tool, and contains the reported output of that run.
+    */
+  trait Run private[sarif] {
+
+    /** @return
+      *   Information about the tool or tool pipeline that generated the results in this run. A run can only contain
+      *   results produced by a single tool or tool pipeline. A run can aggregate results from multiple log files, as
+      *   long as context around the tool run (tool command-line arguments and the like) is identical for all aggregated
+      *   files.
+      */
+    def tool: Tool
+
+    /** @return
+      *   The set of results contained in an SARIF log. The results array can be omitted when a run is solely exporting
+      *   rules metadata. It must be present (but may be empty) if a log file represents an actual scan.
+      */
+    def results: List[Result]
+
+    /** @return
+      *   The artifact location specified by each uriBaseId symbol on the machine where the tool originally ran.
+      */
+    def originalUriBaseIds: Map[String, ArtifactLocation]
+  }
+
+  /** Describes a sequence of code locations that specify a path through a single thread of execution such as an
+    * operating system or fiber.
+    */
+  trait ThreadFlow private[sarif] {
+
+    /** @return
+      *   A temporally ordered array of 'threadFlowLocation' objects, each of which describes a location visited by the
+      *   tool while producing the result.
+      */
+    def locations: List[ThreadFlowLocation]
+  }
+
+  /** A location visited by an analysis tool while simulating or monitoring the execution of a program.
+    */
+  trait ThreadFlowLocation private[sarif] {
+
+    /** @return
+      *   The code location.
+      */
+    def location: Location
+  }
+
+  trait Tool private[sarif] {
+    def driver: ToolComponent
+  }
+
+  trait ToolComponent private[sarif] {
+    def name: String
+    def fullName: String
+    def organization: String
+    def semanticVersion: String
+    def informationUri: URI
+  }
+
+  object Level {
+    val None    = "none"
+    val Note    = "note"
+    val Warning = "warning"
+    val Error   = "error"
+
+    def cvssToLevel(cvssScore: Double): String = {
+      cvssScore match {
+        case score if score == 0.0  => None
+        case score if score <= 3.9  => Note
+        case score if score <= 6.9  => Warning
+        case score if score <= 10.0 => Error
+        case score =>
+          logger.error(s"Score '$score' is not a valid CVSS score! Defaulting to 'warning' SARIF level.")
+          Warning
+      }
+    }
+
+  }
+
+  val serializers: List[Serializer[?]] = List(
+    new CustomSerializer[SarifSchema.Sarif](implicit format =>
+      (
+        { case _ =>
+          ???
+        },
+        { case sarif: SarifSchema.Sarif =>
+          Extraction.decompose(Map("version" -> sarif.version, "$schema" -> sarif.`$schema`, "runs" -> sarif.runs))
```

Comment:
Why do we need the ugly `$schema` member if we have this custom decompose step which can do and already has a mapping?